app/jobs/create_institution_records_job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+class CreateInstitutionRecordsJob < ApplicationJob
+  queue_as :default
+
+  def perform(institution_type_id)
+    i = 1
+    loop do
+      rspo_api_uri = URI("http://194.54.26.132/api/placowki/?page=#{i}&typ_podmiotu_id=#{institution_type_id}")
+      request = Net::HTTP::Get.new(rspo_api_uri)
+      request['accept'] = 'application/json'
+      response = Net::HTTP.start(rspo_api_uri.hostname, rspo_api_uri.port) do |http|
+        http.request(request)
+      end
+
+      raw_institutions = JSON.parse(response.body)
+
+      # the records are sent in pages of 100
+      break if raw_institutions == []
+
+      i += 1
+
+      batch_insert_institutions = []
+      raw_institutions.each do |raw_institution|
+        institution = {}
```

Comment:
You can slightly shorten it in an idiomatic way using each_with_object
```suggestion
      raw_institutions.each_with_object({}) do |raw_institution, institution|
```